--- cthon04.orig/tests.init	2018-11-07 16:33:59.543838497 +0800
+++ cthon04/tests.init	2018-11-07 16:33:11.210474462 +0800
@@ -217,12 +217,6 @@
 # Use with Linux 2.6 / gcc 4.0
 CFLAGS=`echo -DLINUX -DHAVE_SOCKLEN_T -DGLIBC=22 -DMMAP -DSTDARG`
 LIBS=`echo -lnsl`
-
-# Use with Linux glibc < 2.26
-#CFLAGS="-DLINUX -DHAVE_SOCKLEN_T -DGLIBC=22 -DMMAP -DSTDARG "
-#CFLAGS+=`pkg-config --cflags libtirpc`
-#LIBS=`pkg-config --libs libtirpc`
-
 MOUNT=/bin/mount
 UMOUNT=/bin/umount
 LOCKTESTS=`echo tlocklfs tlock64`
