--- tests.init.orig     2018-08-16 11:27:35.109934875 +0800
+++ tests.init  2018-08-16 11:30:03.977145053 +0800
@@ -215,8 +215,14 @@ UMOUNT=/bin/umount
 LOCKTESTS=`echo tlocklfs tlock64`

 # Use with Linux 2.6 / gcc 4.0
-CFLAGS=`echo -DLINUX -DHAVE_SOCKLEN_T -DGLIBC=22 -DMMAP -DSTDARG`
-LIBS=`echo -lnsl`
+#CFLAGS=`echo -DLINUX -DHAVE_SOCKLEN_T -DGLIBC=22 -DMMAP -DSTDARG`
+#LIBS=`echo -lnsl`
+
+# Use with Linux glibc > 2.26
+CFLAGS="-DLINUX -DHAVE_SOCKLEN_T -DGLIBC=22 -DMMAP -DSTDARG "
+CFLAGS+=`pkg-config --cflags libtirpc`
+LIBS=`pkg-config --libs libtirpc`
+
 MOUNT=/bin/mount
 UMOUNT=/bin/umount
 LOCKTESTS=`echo tlocklfs tlock64`
