From cc4a63bfc4c90aa31f9b373424bba325d8091259 Mon Sep 17 00:00:00 2001
From: Jan Stancek <jstancek@redhat.com>
Date: Tue, 19 Jun 2018 13:00:54 +0200
Subject: [PATCH] open_posix: add LTP_ATTRIBUTE* macros to supress warnings

Signed-off-by: Jan Stancek <jstancek@redhat.com>
Signed-off-by: Ping Fang <pifang@redhat.com>
---
 testcases/open_posix_testsuite/include/posixtest.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/testcases/open_posix_testsuite/include/posixtest.h b/testcases/open_posix_testsuite/include/posixtest.h
index 8e704b6ff..dd1a9b291 100644
--- a/testcases/open_posix_testsuite/include/posixtest.h
+++ b/testcases/open_posix_testsuite/include/posixtest.h
@@ -43,3 +43,7 @@
 #ifndef ARRAY_SIZE
 #define ARRAY_SIZE(arr) (sizeof(arr) / sizeof(arr[0]))
 #endif
+
+#define LTP_ATTRIBUTE_NORETURN		__attribute__((noreturn))
+#define LTP_ATTRIBUTE_UNUSED		__attribute__((unused))
+#define LTP_ATTRIBUTE_UNUSED_RESULT	__attribute__((warn_unused_result))
-- 
2.14.4

