diff --git a/testcases/kernel/tracing/dynamic_debug/dynamic_debug01.sh b/testcases/kernel/tracing/dynamic_debug/dynamic_debug01.sh
index f39d67d0a..7f06c2488 100755
--- a/testcases/kernel/tracing/dynamic_debug/dynamic_debug01.sh
+++ b/testcases/kernel/tracing/dynamic_debug/dynamic_debug01.sh
@@ -127,7 +127,7 @@ ddebug_test()
        sed -i -e 1,`wc -l < ./dmesg.old`d ./dmesg.new
        if grep -q -e "Kernel panic" -e "Oops" -e "general protection fault" \
                -e "general protection handler: wrong gs" -e "\(XEN\) Panic" \
-               -e "fault" -e "warn" -e "BUG" ./dmesg.new ; then
+               -e "fault" -e "warn" -e "\<BUG\>" ./dmesg.new ; then
                tst_res TFAIL "Issues found in dmesg!"
        else
                tst_res TPASS "Dynamic debug OK"
