diff --git a/testcases/kernel/mem/lib/mem.c b/testcases/kernel/mem/lib/mem.c
index a0c1b9b00..0833fd602 100644
--- a/testcases/kernel/mem/lib/mem.c
+++ b/testcases/kernel/mem/lib/mem.c
@@ -77,6 +77,8 @@ static void child_alloc(int testcase, int lite, int threads)
 	int i;
 	pthread_t *th;
 
+	FILE_PRINTF("/proc/self/oom_adj", "10");
+
 	if (lite) {
 		int ret = alloc_mem(TESTMEM + MB, testcase);
 		exit(ret);
